Frameworks/UIKit/_UIGestureCoordinator.mm

```diff
@@ -0,0 +1,457 @@
+            if (recognized && DEBUG_GESTURES) {
+                TraceVerbose(TAG, L"Gesture (%hs) handled.", object_getClassName(curgestureClass));
+            }
+
+            if ([curGesture isKindOfClass:[UIPanGestureRecognizer class]]) {
+                // TODO: moved from UIPanGestureRecognizer, it is unclear why this special treatment is required earlier
+                // or should not this be the common treatment for all gestures?
+                shouldCancelTouches |= curGesture.cancelsTouchesInView;
+            } else {
+                shouldCancelTouches |= recognized;
+            }
+        }
+    }
+
+    // Gesture has not been recognized, fire Dimanip Gestures.
+    if (!recognized) {
+        for (UIGestureRecognizer* curGesture in _dManipRecognizers) {
+            id curgestureClass = [curGesture class];
+            if ([curGesture isKindOfClass:[_UIDMPanGestureRecognizer class]]) {
+                BOOL recognized = [curgestureClass _fireGesture:curGesture];
+                if (recognized && DEBUG_GESTURES) {
+                    TraceVerbose(TAG, L"Gesture (%hs) handled.", object_getClassName(curgestureClass));
+                }
+
+                shouldCancelTouches |= recognized;
+            }
+        }
+    }
+
+    //  Removed/reset failed/done/cancelled gestures, including gestures and dManipGestures
+    [self _resetAndClearGestureInEndingStateFrom:_dManipRecognizers];
+    [self _resetAndClearGestureInEndingStateFrom:_regularRecognizers];
+
+    return shouldCancelTouches;
+}
+
+// given an activeGesture, enumerate possible gesture list
+// cancel gesture that can not concurrently fire and remove it from the intenral possible state gesture list
+// NOTE: it is possible that the cancelled gesture is still in regular gesture list with the state to be cancelled
+// But since we are in the middle of enumerating regular gesture list, remove it is from the list is actually
+// less efficient than just check if it is in cancelled state, so we don't remove it here. it will be removed
+// at end of processingGesture
+- (void)_removeGesturesByActiveGesture:(UIGestureRecognizer*)activeGesture {
+    BOOL activeResponds =
+        [activeGesture.delegate respondsToSelector:@selector(gestureRecognizer:shouldRecognizeSimultaneouslyWithGestureRecognizer:)];
+
+    [_gesturesInPossibleState removeObject:activeGesture];
+    for (UIGestureRecognizer* possibleGesture in _gesturesInPossibleState) {
+        // Based on reference platform: test if active gesture can prevent other gesture, if so, further check if the other gesture
+        // can be prevented by active gesture if both result are YES, we need consult the delegates for both gesture.
+        // if both delegates saying NO for concurrent firing the other gesture will be prevented from firing by active gesture.
+        // NOTE: if delegate is not found, referece platform treat it as if the delegate exists and return NO
```

Comment:
through testing. doc didn't say anything.

---
In reply to: [100627710](https://github.com/Microsoft/WinObjC/pull/1909#discussion_r100627710) [](ancestors = 100627710)